internal/controller/git/git.go

```diff
@@ -89,8 +118,13 @@ type Repo interface {
 	// GetDiffPaths returns a string slice indicating the paths, relative to the
 	// root of the repository, of any new or modified files.
 	GetDiffPaths() ([]string, error)
```

Comment:
This is not used anywhere. Mind getting rid of it (and its implementation) while you're already touching this file?